osu.Game/Screens/Backgrounds/BackgroundScreenBeatmap.cs

```diff
@@ -35,18 +52,51 @@ public WorkingBeatmap Beatmap
                             Background.FadeOut(250);
                             Background.Expire();
                         }
-
                         b.Depth = newDepth;
-                        AddInternal(Background = b);
+                        FadeContainer.Child = Background = b;
+                        InternalChild = FadeContainer;
                         Background.BlurSigma = BlurTarget;
                     }));
                 });
             }
         }
 
+        public override void OnEntering(IScreen last)
+        {
+            base.OnEntering(last);
+            DimLevel.ValueChanged += _ => updateBackgroundDim();
+            UpdateDim.ValueChanged += _ => updateBackgroundDim();
+            updateBackgroundDim();
+        }
+        public override void OnResuming(IScreen last)
+        {
+            base.OnResuming(last);
+            updateBackgroundDim();
+        }
+
+        public override bool OnExiting(IScreen last)
+        {
+            UpdateDim.Value = false;
+            return base.OnExiting(last);
+        }
+
+        private void updateBackgroundDim()
+        {
+            if (UpdateDim)
+                FadeContainer?.FadeColour(OsuColour.Gray(BackgroundOpacity), 800, Easing.OutQuint);
+            else
+                FadeContainer?.FadeColour(OsuColour.Gray(1.0f), 800, Easing.InQuint);
```

Comment:
The easing logic was changed inadvertently?

```suggestion
                FadeContainer?.FadeColour(OsuColour.Gray(1.0f), 800, Easing.OutQuint);
```

Reply:
```suggestion
                FadeContainer?.FadeColour(Color4.White, 800, Easing.InQuint);
```